### PR TITLE
Bump geopandas from 0.12.1 to 0.13.2, remove pygeos

### PR DESCRIPTION
**Pull request by weiji14**
Geopandas 0.13.x comes with better support for the newer GeoParquet 1.0.0-beta.1 spec, that may help with the benchmarking efforts happening at the ICESat-2 Hackweek 2023 h5cloud project - https://github.com/ICESAT-2HackWeek/h5cloud

Bumps [geopandas](https://github.com/geopandas/geopandas) from 0.12.1 to 0.13.2.
- [Release notes](https://github.com/geopandas/geopandas/releases)
- [Changelog](https://github.com/geopandas/geopandas/blob/main/CHANGELOG.md)
- [Commits](https://github.com/geopandas/geopandas/compare/v0.12.1...v0.13.2)

### CODE REVIEW

**Comment by weiji14** [comment]
If there's no rush, I'd like to wait for `dask-geopandas` to drop `pygeos` in conda-forge - https://github.com/conda-forge/dask-geopandas-feedstock/issues/12, so that we don't get that annoying warning like:

```
UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
```

File: conda-linux-64.lock [hunk 1]
```diff
@@ -417,7 +416,6 @@ https://conda.anaconda.org/conda-forge/linux-64/psycopg2-2.9.6-py310h76c1b15_0.c
 https://conda.anaconda.org/conda-forge/linux-64/pulseaudio-client-16.1-hb77b528_4.conda#8f349ca16d30950aa00870484d9d30c4
 https://conda.anaconda.org/conda-forge/noarch/pyasn1-modules-0.2.7-py_0.tar.bz2#ad1e886d09700b2304975335f714bd9c
 https://conda.anaconda.org/conda-forge/noarch/pyct-core-0.4.6-py_0.tar.bz2#55ec526f95e0959de3f68bc6289a20da
-https://conda.anaconda.org/conda-forge/linux-64/pygeos-0.14-py310h056c13c_1.conda#047c24638f8c34b34ee086ddde230bb6
```
Comment (weiji14) [inline conda-linux-64.lock:420]: Ok, `pygeos` is removed now!

**Review by MattF-NSIDC — APPROVED** [review]
LGTM! Excited to hear about new geopandas releases :)

**Pull request merged**